reana_client/cli/files.py

```diff
@@ -341,55 +348,72 @@ def upload_files(ctx, workflow, filenames, access_token):  # noqa: D301
             )
             sys.exit(1)
 
+        try:
+            with open(".gitignore") as f:
+                gitignore_spec = pathspec.PathSpec.from_lines("gitwildmatch", f)
+            display_message(
+                ".gitignore file is detected and will be respected.", msg_type="success"
+            )
+        except FileNotFoundError:
+            gitignore_spec = SimpleNamespace()
+            gitignore_spec.match_file = lambda _: False
```

Comment:
What do you think about setting `gitignore_spec = None` and checking it afterwards? By doing it this way, we can avoid using `SimpleNamespace` and the code should be easier to understand

Reply:
Makes sense. But, instead of `None`, I will use `try/catch/else` cause it will be more pythonic:

```python
try:
    with open(".gitignore") as f:
        gitignore_spec = pathspec.PathSpec.from_lines("gitwildmatch", f)
    display_message(
        ".gitignore file is detected and will be respected.", msg_type="success"
    )
except FileNotFoundError:
    filenames += directories_full_path
else:
    for directory_path in directories_full_path:
        for root, dirs, files in os.walk(directory_path):
            if not gitignore_spec.match_file(root):
                for filename in files:
                    filename_full_path = os.path.join(root, filename)
                    if not gitignore_spec.match_file(filename_full_path):
                        filenames.append(filename_full_path)
```